python/test/test_api.py

```diff
@@ -357,6 +357,11 @@ def test_stop(self):
                     {
                         "backend": "python",
                         "parameters": {"decoupled": {"string_value": "False"}},
+                        # Keep instance count low for fast startup/cleanup.
+                        # Alternatively can use KIND_CPU here, but keeping gpus/count explicit.
+                        "instance_group": [
+                            {"kind": "KIND_GPU", "gpus": [0], "count": 1}
```

Comment:
torn on this - I guess we don't have a final recommendation on exit timeout per instance .... 

if we go with default timeout - then I say this change makes sense -

if we set it explicitly - then I'm tempted to say the test could discover the number of GPUs and set the exit timeout to some multiple, say 2 * NUM_GPUS

Reply:
Similar to other coment, I think adding some fine-grained tests with expected exit timeouts to track potential regressions in server cleanup/shutdown expectations could be valuable in future - but just aiming to get a consistent pass for CI with this change for now. We can expand on things like number of instances, number of GPUs, and tweaking timeout accordingly as we some more time to improve the tests.